Added ⚡️ content to `Payments` >  `Sending bitcoin` (#589)
BitcoinDesign/Guide: guide/payments/send.md

# Sending bitcoin

> Sending bitcoin can be a very simple, or a complex flow in a Bitcoin application. People may want to send bitcoin to a known contact, transfer it to another wallet on a different device, or make a purchase via a [payment processor]({{ '/guide/getting-started/software/#payment-processors' | relative_url }}).

Reviewer: ```suggestion
Sending bitcoin can be a very simple, or complex flow in a Bitcoin application. People may want to send bitcoin to a known contact, transfer it to another wallet on a different device, or make a purchase via a [payment processor]({{ '/guide/getting-started/software/#payment-processors' | relative_url }}).
```

## Payment entry points

The need to send bitcoin can be triggered by many different use cases, and initiated by both sender and receiver. Common entry points are:

> - Manual entry of address and amount

Reviewer: Does this mean manually typing in an address word for word? Got a bit confused on this point.

Author: Yes. Could also be manual pasting. It's the most basic form of input and ideally just a fallback that users never have to resort to.

- Sending to a previously saved contact

> - Copy & pasting payment information

Reviewer: ```suggestion
- Copy & pasting a payment request
```

Author: Doesn't have to be a request, could also be just a Lightning address.

> - Clicking a payment link

Reviewer: ```suggestion
- Clicking a payment request link
```

Author: Why add this?

Reviewer: It's more descriptive of what it is

> - Scanning a QR code

Reviewer: ```suggestion
- Scanning a payment request QR code
```

Author: Could also scan a QR code for a Lightning address.

Author: Unlikely though. Point is that it could also just be recipient info that you want to save to your address book.

Reviewer: Maybe I'm getting too technical but a Lightning address is still essentially a payment request it just does it behind the scenes.

> - Holding a device close to another device (using [NFC](https://en.wikipedia.org/wiki/Near-field_communication))

Reviewer: ```suggestion
- Receiving a payment request over ([NFC](https://en.wikipedia.org/wiki/Near-field_communication))
```

Author: Should this also include BlueTooth for completeness?

Reviewer: Could just say something like 'Receive a payment request with a contact less payment (NFC or BlueTooth).

<div class="image-slide-gallery">

{% for item in page.imagesEntryPoints %}

{% capture imageURL %}{{ page.image_base }}{{ item.file }}.png{% endcapture %}
{% capture imageURLRetina %}{{ page.image_base }}{{ item.file }}@2x.png{% endcapture %}

{% include picture.html
   image = imageURL
   retina = imageURLRetina
   alt-text = item.alt
   caption = item.caption
   width = 250
   height = 541
   layout = "shadow"
%}
{% endfor %}

</div>

> Since users cannot control how an address or invoice is presented to them, wallets should be highly flexible in terms of input options and interoperability.

Reviewer: ```suggestion
Since users cannot control how a payment request is presented to them, wallets should be highly flexible in terms of input options and [interoperability]({{ '/guide/designing-products/wallet-interoperability/' | relative_url }}). If your wallet does not support a particular payment request, users should be presented with an human readable [error]({{ '/guide/payments/send/#errors' | relative_url }}).
```

Author: Small typo in the suggestion, should be `a human readable` (not `an`). Would you mind updating, then I'll commit?

Reviewer: Updated :)

Payment information can be shared in many formats and over diverse communication channels. Each has its own advantages and limitations. More details on the [payment request formats]({{ '/guide/payments/send/payment-request-formats/' | relative_url }}) page.

## Presenting payment requests

> Once the application has registered the payment request, it should provide clear instructions and options to the user.

Reviewer: ```suggestion
Once the application has imported a payment request, it should provide clear instructions and options to the user.
```

Registered sounded funny to me.

<div class="image-slide-gallery">

{% for item in page.imagesEntryScreens %}

{% capture imageURL %}{{ page.image_base }}{{ item.file }}.png{% endcapture %}
{% capture imageURLRetina %}{{ page.image_base }}{{ item.file }}@2x.png{% endcapture %}

{% include picture.html
   image = imageURL
   retina = imageURLRetina
   alt-text = item.alt
   caption = item.caption
   width = 250
   height = 541
   layout = "shadow"
%}
{% endfor %}

</div>

> ## Manual payment initiation

Reviewer: ```suggestion
## Manual payment configuration
```

The payment has already been initiated by the requester at this stage, configuration seemed more appropriate.

Author: Actually, this is a parallel route in which no payment has been initiated, but the user does this on their own.

> When responding an invoice that contains all relevent information, the user can quickly review and approve it. In other scenarios, it may be required to manually enter or edit various details.

Reviewer: ```suggestion
In many situations payment requests will come pre-filled with the payment details for the sender. In some situations however users may need to manually configure some options.
```

Wasn't quite clear to me before what this was saying.

<div class="image-slide-gallery">

{% for item in page.imagesInfo %}

{% capture imageURL %}{{ page.image_base }}{{ item.file }}.png{% endcapture %}
{% capture imageURLRetina %}{{ page.image_base }}{{ item.file }}@2x.png{% endcapture %}

{% include picture.html
   image = imageURL
   retina = imageURLRetina
   alt-text = item.alt
   caption = item.caption
   width = 250
   height = 541
   layout = "shadow"
%}
{% endfor %}

> </div>

Reviewer: I'd recommend adding a 'note' and 'tags' sub-sections to this section and explain them in more detail. Fee's could also be briefly mentioned as it relates to manual configuration then link to the next section on fees for the deeper dive.

Author: Added a "meta-data" section with a bit of background on why this is useful.

Reviewer: Yep great addition!

> **Address**

Reviewer: This is a bit confusing to me, I think the term address should be reserved for an on-chain address. Also Lightning node ID sending should be one of those advanced things we don't cover in the meat of the content imo. 

How does this relate to 'Manual payment initiation'? Seems to be just a descriptive entry on what an address is not anything about manual configuration.

Reviewer 2: Maybe an "identifier" instead of an "address"?

Author: How about "Recipient" or "To" (same as email)?

Main thing is that there's a human-readable label that lets people understand that they need to enter some type of destination for those bitcoin. That recipient can be a contact, static invoice, on-chain address, etc.

It relates to `Manual payment initiation` as you have to specify who gets the bitcoin when you want to make a payment (unless you want to burn them).

Author: Switched this to "Recipient" and rewrote the copy. Please let me know if it's more clear now.

Also added a Home screen with a caption to clarify that this is a user flow where users initiate the payment in the app and need to manually enter all info.

Reviewer: Yep flowed a lot better and this section clicked much faster for me with this change.

An address is a piece of information that bitcoin can be sent to, and that doesn’t change. Here are the common formats:
- On-chain address
- Lightning node ID
- Lightning address

> Not that there are also static [invoice types]({{ '/guide/payments/send/payment-request-formats/' | relative_url }}) that can receive payments repeatedly. These are less intuitive overall due to their appearance, but could also be considered payment endpoints.

Reviewer: ```suggestion
There are also static [invoice types]({{ '/guide/payments/send/payment-request-formats/' | relative_url }}) that can receive payments repeatedly. These are less intuitive overall due to their appearance, but could also be considered payment endpoints.
```

> **Amounts**

Reviewer: ```suggestion
**Amount**
```

If no amount is provided via a payment request, manual entry should be simple and convenient so users don’t accidentally send an incorrect amount. The amount should be displayed in both bitcoin or satoshi value, as well as the user’s local currency. Options to quickly toggle between them should be available. More on the [Units & Symbols page]({{ '/guide/payments/units-and-symbols/' | relative_url }}).

> ## Fees

Reviewer: This section should have some content around how a user could interact with fees (changing them for on-chain for example).

Author: Added a visual for a fee picker.

Reviewer: Yew, looks good!

Payment fees can drastically differ based on a few attributes:

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/payments/send/fee-options.png"
   retina = "/assets/images/guide/payments/send/fee-options@2x.png"
   alt-text = "Example image"
   width = 400
   height = 417
   layout = "float-right-desktop"
%}

> **Lightning network routing fees**

Reviewer: ```suggestion
**Lightning routing fees**
```

On the Lightning network, payments are passed between nodes to get from the sender to the receiver. Each of those nodes may charge a base fee and a second fee based on a percentage of the amount forwarded. Fees paid can vary, but are typically in the single-digit or double-digit Satoshi range (a small fraction of on-chain fees).

**Lightning service provider fees**

> In certain situations, the Lightning wallet may not have enough liquidity to send or receive a payment. Wallet providers may offer to alleviate these friction points, and earn additional fees. A common scenario is the automatic opening of a payment channel when a wallet receives the first deposit.

Reviewer: ```suggestion
In certain situations, the Lightning wallet may not have enough channel liquidity to send a payment. Wallet providers may offer to alleviate these friction points, and earn additional fees. A common scenario is the automatic opening of a payment channel when a wallet attempts to send a payment larger than their outbound capacity.
```

This should stay contextual to sending.

**On-chain fees**

This fee is dependent on how many other transactions are currently waiting to be processed on the base layer as a whole. The [average fee](https://ycharts.com/indicators/bitcoin_average_transaction_fee) in January 2021 was $0.63, and $28.60 in April 2021.

</div>

## Review & approval

<div class="center" markdown="1">

{% include image.html
   image = "/assets/images/guide/payments/send/confirm.png"
   retina = "/assets/images/guide/payments/send/confirm@2x.png"
   alt-text = "Example image"
   caption = "A compact summary to confirm the information is accurate."
   width = 250
   height = 541
   layout = "float-right-desktop -background -shadow"
%}

> Particularly with payments for larger amounts, it is good practice to allow users to review and confirm the information before it is sent.

Reviewer: I think this should be for any payment amount, bitcoin payments are final once sent and should never skip some kind of review and confirm.

Author: Might need to rephrase this. When you just clicked a Lightning invoice for 10 sats, all you do is review and hit "Pay". That's where no additional review screen after the invoice presentation is necessary (IMHO), it would just be duplication.

Reviewer: It would just go straight to a review page wouldn't it? So it still has the same final screen for any payment amount.

Author: I think we're saying the same thing?

Payments initiated via invoices which don't require any additional input by the user can avoid this review step, as the whole user interaction is one of review and confirmation.

> Wallets that offer features for [pending limits]({{ '/guide/onboarding/protecting-a-wallet/#wallet-limits' | relative_url }})s or support [multi-key]({{ '/guide/private-key-management/multi-key/' | relative_url }}) schemes for additional security may ask users to go through additional confirmation steps here.

Reviewer: ```suggestion
Wallets that offer features for [payment limits]({{ '/guide/onboarding/protecting-a-wallet/#wallet-limits' | relative_url }}) or support [multi-key]({{ '/guide/private-key-management/multi-key/' | relative_url }}) schemes for additional security may ask users to go through additional confirmation steps here.
```
Should this be payment limits? I don't think we should mention multi-key here, the wallet product this section is covering should be LN focused which would not encompass a multi-key element.

Reviewer 2: ```suggestion
Wallets that offer features for [spending limits]({{ '/guide/onboarding/protecting-a-wallet/#wallet-limits' | relative_url }}) or support [multi-key]({{ '/guide/private-key-management/multi-key/' | relative_url }}) schemes for additional security may ask users to go through additional confirmation steps here.
```

Author: @Bosch-0 it was meant to be spending limits.

</div>

> ## Transaction processing

Reviewer: ```suggestion
## Payment processing
```

I notice we keep jumping between using the term payment and transaction. As this section is called payments I'd stick with calling them Payments.

Author: I use them pretty much interchangeably. Just using `Payments` can appear repetitive. 

I am confused about this anyways since reading @sbddesign's distinction that transactions are on-chain things and payments are Lightning things. We should clarify this.

Reviewer: To me anyways transactions are much more complex things in bitcoin - a payment is much more simple and aligned with what an end user would be concerned with

Author: Asked about this in the #bitcoin-design-guide channel to get broader input.

> Processing times may also differ a lot between on-chain and Lightning network payments. On-chain, pending transactions are bundled into a [new block]({{ '/guide/getting-started/technology-primer/#what-is-a-blockchain' | relative_url }}) roughly every 10 minutes. On the Lightning network, [payment routing]({{ '/guide/getting-started/technology-primer/#how-are-payments-routed' | relative_url }}) happens instantly and is largely dependent on the number of nodes involved and their responsiveness.

Reviewer: ```suggestion
Payment processing times may differ between on-chain and the Lightning network. On-chain, pending transactions are bundled into a [new block]({{ '/guide/getting-started/technology-primer/#what-is-a-blockchain' | relative_url }}) roughly every 10 minutes. On the Lightning network, [payment routing]({{ '/guide/getting-started/technology-primer/#how-are-payments-routed' | relative_url }}) happens instantly and is largely dependent on the number of nodes involved, their liquidity, and payment processing speeds.
```

Author: The term `payment` already appears 41 times on the page, I'd prefer to to add more instances if it isn't needed.

When transactions take longer than expected, users need to be clearly informed about the status. In scenarios like in-store payments, speedy confirmation is of the essence, as the user wants to move on, and the merchant may have other customers waiting. In-app status updates can be coupled with notifications, or even emails, to ensure that both parties are confident that everything is in order.

<div class="image-slide-gallery">

{% for item in page.imagesProcessing %}

{% capture imageURL %}{{ page.image_base }}{{ item.file }}.png{% endcapture %}
{% capture imageURLRetina %}{{ page.image_base }}{{ item.file }}@2x.png{% endcapture %}

{% include picture.html
   image = imageURL
   retina = imageURLRetina
   alt-text = item.alt
   caption = item.caption
   width = 250
   height = 541
   layout = "shadow"
%}
{% endfor %}

</div>

In case a user needs to briefly wait, the application should not block the interface, but offer other useful actions to perform, such as labelling and tagging the payment, or adding the recipient to the contact list.

> On-chain, it may also be possible to cancel (via [replace-by-fee](https://bitcoinops.org/en/topics/replace-by-fee/) or double-spend with a higher fee) or speed-up (via [child-pays-for-parent](https://bitcoinops.org/en/topics/cpfp/)) a transaction after it has been broadcast, but before it has been included in a block.

Reviewer: ```suggestion
On-chain, you may want to offer your users the option to cancel (via [replace-by-fee (RBF)](https://bitcoinops.org/en/topics/replace-by-fee/)) or speed-up (via [child-pays-for-parent (CPFP)](https://bitcoinops.org/en/topics/cpfp/)) a transaction after it has been broadcast, but before it has been included in a block.
```

With points like this I think we should speak to the reader like above. Does double spending cancel or does the higher fee just get confirmed?

Author: `or double-spend with a higher fee` is an explanation of what `replace by fee` does. You just make another transaction with the same coins but a higher fee, so it gets confirmed faster than the first one. The first one remains in the mempool, but will be rejected by the network because the coins have then been spent by the second transaction.

I may need to revise my writing with this active language that directly addresses the reader. Have mostly tried to stay away from it.

Author: Rewrote this part slightly. We should add RBF and CPFP to the glossary with simple descriptions, and link to there.

Author: Created an issue for adding the two terms to the Glossary.

- How the problem can be fixed or avoided
- What to do next

<div class="image-slide-gallery">

{% for item in page.imagesErrors %}

{% capture imageURL %}{{ page.image_base }}{{ item.file }}.png{% endcapture %}
{% capture imageURLRetina %}{{ page.image_base }}{{ item.file }}@2x.png{% endcapture %}

{% include picture.html
   image = imageURL
   retina = imageURLRetina
   alt-text = item.alt
   caption = item.caption
   width = 250
   height = 541
   layout = "shadow"
%}
{% endfor %}

</div>

It is ideal when the application can automatically identify and fix or avoid the problem. This may not always be possible, or even wanted if the solution incurs extra fees or takes time. For practical purposes, it can be helpful to group problems into categories, such as:

- The problem cannot be identified
- The problem can be identified and requires manual action by the user
- The problem can be automatically fixed, but the user needs to choose which solution to use
- The app can likely automatically fix the problem with negligible impact on the user

Effectively supporting users when problems occur can build trust and confidence, and essential aspect for financial applications.

---

> ## Advanced options

Reviewer: Some things you could add here:

- RBF
- Let users choose payment hops
- Enable an AMP payment (this should be by default though, might want to add something about this in this section as it's a major UX improvement).

Author: RBF is already mentioned, or do you mean I should provide more detail (like a mock-up)?

By payment hops, is that the discussion around probability of success and fees? Or what am I choosing?

AMP seems like an under-the-hood thing, so I only included it in the format sub-page. Ideally, the wallet can pick the most appropriate technique for making the payment/transaction.

# Sending bitcoin

Sending bitcoin is one of the most essential user activities in a Bitcoin application, and also one of the least structured ones. People may want to send bitcoin to a known contact, transfer it to another wallet on a different device, or make a purchase via a [payment processor]({{ '/guide/getting-started/software/#payment-processors' | relative_url }}).

> ## Payment entry points

Reviewer: Another payment entry point you may want to consider adding is generating and sharing a withdraw request. I'm covering this in my requesting bitcoin page. By withdraw request I'm referring to LNURL-withdraw (withdraw request sounds better imo). 

You could expand on this somewhere in this section and discuss how these could be used to generate P2P gift cards.

Author: I was thinking we could have a "Withdrawing bitcoin" page at a later point.

Reviewer: Would there be enough content for another dedicated page? Could easily be covered across sending / request page imo

Reviewer 3: I don't think it would be enough for a dedicated page. It's def a use-case we should cover, perhaps in a follow up? One more example besides a gift would be a user getting a refund from a merchant so they need to scan a QR in order to pull the funds out.

The need to send bitcoin can be triggered by many different use cases, and initiated by both sender and receiver. Common entry points are:

> - Manual entry of address and amount

Reviewer 3: What's the difference between:
- Manual entry of address and amount and 
- Copy & pasting payment information?

Author: Manual entry is the most basic entry form and has to be supported.

Copy & paste can be an interaction with an input field, but ideally, the app auto-detects clipboard content and presents the appropriate UI automatically.

Reviewer: Does it have to be supported though? I can't imagine entering it like that and if they did it has major room for error.
